include/ofi_shm.h

```diff
@@ -227,9 +230,24 @@ struct smr_inject_buf {
 	};
 };
 
+enum {
+	SMR_SAR_FREE = 0, /* buffer can be used */
+	SMR_SAR_READY, /* buffer has data in it */
+};
+
+struct smr_sar_buf {
+	uint64_t	status;
+	uint8_t		buf[SMR_SAR_SIZE];
+};
```

Comment:
What do you think is the best way to store this status? Have the SAR_SIZE be smaller to align the whole buffer?

Reply:
I don't have a great answer here.  Unless buf is aligned and a multiple of page size, copy routines are going to get out of whack with either the src or dst buffer.  So, I wouldn't make any changes to this now.

Reply:
I can also include this in the peer data once I refactor the addressing/per peer data. Since a peer only uses one SAR buffer at a time, we could just look up the SAR status for that peer.

Reply:
Btw, the buffer pool will throw off all alignment unless it is specified as part of the pool attributes.  So, we can ignore this for quite a while I think.